src/oscar/apps/catalogue/product_attributes.py

```diff
@@ -66,4 +66,13 @@ def save(self):
         for attribute in self.get_all_attributes():
             if hasattr(self, attribute.code):
                 value = getattr(self, attribute.code)
+
+                # only go and save values that have changed, don't do anything useless
```

Comment:
This check already exists in the ``` _save_value``` method as well:
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/catalogue/abstract_models.py#L927

So the comment doesn't really describe what it's actually fixing, so I think it's better to explain the issue here what you've put in the description of this PR: "This makes sure attributes on the parent are not needlessly saved to the child."

The solution seems good to me.

Reply:
You forgot to mention I should add a test too ;)